openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -3083,12 +3113,17 @@ def create_force(self, system, topology, **kwargs):
             # Make a temporary copy of ref_mol to assign charges from charge_mol
             temp_mol = FrozenMolecule(ref_mol)
 
-            # If the molecule wasn't assigned parameters from a manually-input charge_mol, calculate them here
-            temp_mol.generate_conformers(n_conformers=10)
-            temp_mol.compute_partial_charges(quantum_chemical_method=self._quantum_chemical_method,
-                                             partial_charge_method=self._partial_charge_method)
+            try:
+                # If the molecule wasn't assigned parameters from a manually-input charge_mol, calculate them here
+                temp_mol.generate_conformers(n_conformers=self.number_of_conformers)
+                temp_mol.assign_partial_charges(partial_charge_method=self.partial_charge_method)
```

Comment:
Good call, and great attention to detail. I actually think this kwarg is pretty misleading, because it's not propagated to `find_smarts_matches` or any other toolkit-dependent functionality (including `compute_partial_bond_orders`!). But we already did it for `ToolkitAM1BCCHandler`, and we should continue the pattern, so I'll accept this patch. We should do what we can to expand this keyword out to all reasonable toolkit-dependent functionality in the future.